### ISSUE TITLE
mc throws 503 Service Unavailable when preexisting file is used

### ISSUE THREAD

**Issue by balamurugana** [minio/minio#1572]
When running `minio server /mnt/sd{b..e}1/bala/mydisk` where `/mnt/sde1/bala/mydisk/mybucket/foo` is present already, below step causes `503 Service Unavailable`
1. `$ mc mb myminio/mybucket`
2. `$ mc cat myminio/mybucket/foo` throws `503 Service Unavailable`

I think the expected behaviour is `NoSuchKey` error.

**Comment by balamurugana** [comment]
mc debug output

```
mc: <DEBUG> GET /mybucket/?location= HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015509Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 200 OK
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:09 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 7ZGAT5FUWSVNYKC8

mc: <DEBUG> Response Time:  777.339µs

mc: <DEBUG> GET /mybucket/foo HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015509Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:09 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: GFBMNUUXUP2Y3KPH

10b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>ReadQuorum</Code><Message>Multiple disk failures, unable to reconstruct data.</Message><Key></Key><BucketName></BucketName><Resource>/mybucket/foo</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  598.685µs

mc: <DEBUG> GET /mybucket/foo HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015510Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:10 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 9HDO7GE0UHY7G4O5

10b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>ReadQuorum</Code><Message>Multiple disk failures, unable to reconstruct data.</Message><Key></Key><BucketName></BucketName><Resource>/mybucket/foo</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  759.145µs

mc: <DEBUG> GET /mybucket/foo HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015510Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:10 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 6N6RWD2XCBGELMT2

10b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>ReadQuorum</Code><Message>Multiple disk failures, unable to reconstruct data.</Message><Key></Key><BucketName></BucketName><Resource>/mybucket/foo</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  859.462µs

mc: <DEBUG> GET /mybucket/foo HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015511Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:11 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 831A6V76Z6XTV78P

10b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>ReadQuorum</Code><Message>Multiple disk failures, unable to reconstruct data.</Message><Key></Key><BucketName></BucketName><Resource>/mybucket/foo</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  754.497µs

mc: <DEBUG> GET /mybucket/foo HTTP/1.1
Host: 127.0.0.1:60000
User-Agent: Minio (linux; amd64) minio-go/1.0.1 mc/2016-05-08T09:42:02Z
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20160510/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160510T015519Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Tue, 10 May 2016 01:55:19 GMT
Server: Minio/DEVELOPMENT.2016-05-08T09-37-27Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: ZQFGK8VIVJ9JSDO5

10b
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>ReadQuorum</Code><Message>Multiple disk failures, unable to reconstruct data.</Message><Key></Key><BucketName></BucketName><Resource>/mybucket/foo</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  699.787µs

mc: <ERROR> Unable to read from ‘myminio/mybucket/foo’. 503 Service Unavailable
 (3) cat-main.go:150 main.mainCat(..) Tags: [myminio/mybucket/foo]
 (2) cat-main.go:92 main.catURL(..) Tags: [myminio/mybucket/foo]
 (1) common-methods.go:70 main.getSourceStreamFromAlias(..) Tags: [myminio, http://127.0.0.1:60000/mybucket/foo]
 (0) client-s3.go:164 main.(*s3Client).Get(..)
 Commit:06ad6c53f7a2 | Release-Tag:DEVELOPMENT.2016-05-08T09-42-02Z | Host:minio3 | OS:linux | Arch:amd64 | Lang:go1.6.2 | Mem:2.0MB/4.9MB | Heap:2.0MB/3.7MB
```

**Comment by harshavardhana** [comment]
> When running minio server /mnt/sd{b..e}1/bala/mydisk where /mnt/sde1/bala/mydisk/mybucket/foo is present already, below step causes 503 Service Unavailable
> 1. $ mc mb myminio/mybucket
> 2. $ mc cat myminio/mybucket/foo throws 503 Service Unavailable
> 
> I think the expected behaviour is NoSuchKey error.

You are saying that the readQuorum here is '1' ?

**Comment by balamurugana** [comment]
Fixed in commit id 409e09c1e55ab448419a3d62cd8a5704312b2172

```
$ minio version
Version: 2016-05-10T13:34:20Z
Release-Tag: DEVELOPMENT.2016-05-10T13-34-20Z
Commit-ID: 409e09c1e55ab448419a3d62cd8a5704312b2172
```
